### DISCUSSION TITLE
Polybar not starting or showing up in HLWM

### DISCUSSION THREAD

**Discussion by regamichael** [polybar/polybar#2648] [Q&A] [answered]
After an Update the polybar didn't show up anymore. I can still launch it from terminal and it doesn't show any errors when starting it. It just doesn't start anymore on reboot...
The HLWM config file is still the same and the polybar file wasn't changed either. Could someone help me pls?

**Comment by patrick96** [comment, marked as answer]
Does polybar print any errors when it starts as part of HLWM's startup routine? You will need to redirect polybar's output to some log file to see its output, have a look at our [debugging guide](https://github.com/polybar/polybar/wiki/Debugging-your-Config).

**Reply by regamichael** [reply]
I solved the problem by just adding: ~/.config/polybar/launch.sh to .bash_profile. Polybar was not showing any errors, I think the problem the sequence of how polybar and hlwm startet.

**Reply by patrick96** [reply]
I would very much discourage starting polybar in any kind of shell startup file.
`.bash_profile` not only runs when you first log in. It runs when you log into the system through ssh and whenever you run bash with the `-l` argument.
In neither case do you want to start polybar.

> Polybar was not showing any errors, I think the problem the sequence of how polybar and hlwm startet.

What was it showing? If it didn't produce any output then either you didn't correctly collect the output produced by polybar or the polybar command was never executed in the first place.
